index.js

```diff
@@ -18,19 +18,28 @@ function getEnvironment (options) {
 exports.name = 'nunjucks';
 exports.outputFormat = 'html';
 
-exports.render = function (str, options, locals) {
+exports.compile = function (str, options) {
   var env = getEnvironment(options)
-  return env.renderString(str, locals)
+  var tmpl = new Template(str, env, true)
+  return function (locals) {
+    return tmpl.render(locals)
+  }
 }
 
-exports.renderAsync = function (str, options, locals) {
-  var env = getEnvironment(options)
-  return new Promise(function (fulfill, reject) {
-    env.renderString(str, locals, function (err, res) {
-      if (err) {
-        return reject(err)
-      }
-      fulfill(res)
+exports.compileAsync = function (str, options) {
+  return new Promise(function (resolve, reject) {
+    var env = getEnvironment(options)
+    var tmpl = new Template(str, env, true)
+    resolve(function (locals) {
+      return new Promise(function (done, fail) {
```

Comment:
…it doesn't work this way.

```js
        compileAsync = (str, options) => Promise => (locals) => output
nunjacksCompileAsync = (str, options) => Promise => (locals) => Promise => output
```

Maybe it'll be better to just change compileAsync's signature to nunjacks'?